src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
 
-}
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid side length');
+    }
 
-export class Circle {
+    const sides = [a, b, c].sort((x, y) => x - y);
 
+    if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error('Invalid triangle sides');
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = 'circle';
+
+  color: string;
+
+  radius: number;
 
+  constructor(color: string, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Invalid radius');
+    }
+
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape = 'rectangle';
+
+  color: string;
+
+  width: number;
+
+  height: number;
+
+  constructor(color: string, width: number, height: number) {
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again